apps/aecore/src/aec_pow_cuckoo.erl

```diff
@@ -219,9 +219,7 @@ wait_for_result(#state{os_pid = OsPid,
                           _                     -> Reason
                       end,
             ?error("OS process died: ~p~n", [Reason2]),
-            {error, {execution_failed, Reason2}};
-        _Other ->
```

Comment:
how we deal with those messages later? Removing this receive clause will make them enqueue in process queue.

Reply:
Such processes will be enqueued only for the duration of the OS process for PoW generation which is a relatively short interval. In practice, do not expect messages in this category. 
If this is a concern, I can
1. Use a different handler for spawn processes that read these messages 
2. Spawn separate erlang process for verification as @lucafavatella proposes below. In this case, however, I have to introduce a new state in the state machine. As I heard refactoring of the miner if planned, I am not sure if this is not an overkill.

Reply:
I see that if any such message is received, it is queued in the aec_miner statem - e.g. see https://github.com/aeternity/epoch/blob/367a0915e1132a80032963a79b26ee7ce23b35c3/apps/aecore/src/aec_miner.erl#L422